File: c/examples/battery_power_cycle.c
#include <stdio.h>
#include "../smart_relay.h"

static int i2c_write_stub(uint8_t addr, const uint8_t *data, uint8_t len) {
  (void)addr; (void)data; (void)len;
  return -1;
}

static int i2c_read_stub(uint8_t addr, uint8_t *data, uint8_t len) {
  (void)addr; (void)data; (void)len;
  return -1;
}

int main(void) {
  smart_relay_t relay = {
    .address = 0x2A,
    .i2c_write = i2c_write_stub,
    .i2c_read = i2c_read_stub
  };

  if (smart_relay_power_cycle_set_max_on_time(&relay, 20) != SMART_RELAY_OK) {
    printf("set max on failed\n");
  }
  if (smart_relay_power_cycle_enable(&relay, 0) != SMART_RELAY_OK) {
    printf("enable power cycle failed\n");
  }

  // Do work here...

  if (smart_relay_power_cycle_sleep(&relay, 300) != SMART_RELAY_OK) {
    printf("sleep command failed\n");
  }

  return 0;
}

File: python/examples/battery_power_cycle.py
"""Battery power-cycle example for Smart Relay I2C module on Linux."""

from smbus2 import SMBus
from smartrelay import SmartRelay
import time


def main():
    with SMBus(1) as bus:
        relay = SmartRelay(bus, address=0x2A)

        relay.power_cycle_set_max_on_time(20)
        relay.power_cycle_enable(0, sleep_enable=True)

        # Do work while power is ON.
        time.sleep(2)

        # Request power-off for 5 minutes.
        relay.power_cycle_sleep(300)


if __name__ == "__main__":
    main()
